File: ocaml/space-age/space_age.ml
open Base

type planet = Mercury | Venus | Earth | Mars
            | Jupiter | Saturn | Neptune | Uranus

let period planet =
  match planet with
  | Mercury -> 0.2408467
  | Venus   -> 0.61519726
  | Earth   -> 1.
  | Mars    -> 1.8808158
  | Jupiter -> 11.862615
  | Saturn  -> 29.447498
  | Uranus  -> 84.016846
  | Neptune -> 164.79132

let secToYears sec = sec /. 31557600.0

let age_on (planet : planet) (sec : int) : float =
  secToYears (Float.of_int sec) /. period planet

File: clojure/space-age/src/space_age.clj
(ns space-age
  (:use [clojure.core]))

(defn on-earth [seconds]
  (/ seconds 31557600.0))

(defn on-mercury [seconds]
  (/ seconds (* 31557600.0 0.2408467)))

(defn on-venus [seconds]
  (/ seconds (* 31557600.0 0.61519726)))

(defn on-mars [seconds]
  (/ seconds (* 31557600.0 1.8808158)))

(defn on-jupiter [seconds]
  (/ seconds (* 31557600.0 11.862615)))

(defn on-saturn [seconds]
  (/ seconds (* 31557600.0 29.447498)))

(defn on-uranus [seconds]
  (/ seconds (* 31557600.0 84.016846)))

(defn on-neptune [seconds]
  (/ seconds (* 31557600.0 164.79132)))
